docs/topic-guides/hp-tuning-det/hp-adaptive-asha.txt

```diff
@@ -51,16 +49,176 @@ Resource budget:
  Details
 *********
 
-Adaptive (ASHA) is an approximation to the resource allocation scheme
-used by :ref:`Adaptive <topic-guides_hp-tuning-det_adaptive-advanced>`.
-While Adaptive promotes hyperparameter configurations synchronously,
-resulting in underutilized nodes waiting on completion of validation
-steps for other configurations, ASHA uses asynchronous promotions to
-maximize compute efficiency of the searcher.
+Conceptually, the ``adaptive_asha`` searcher is a carefully tuned
+strategy for spawning multiple *ASHA* (asynchronous successive halving
+algorithm) searchers, themselves hyperparameter search algorithms. ASHA
+can be configured to make different tradeoffs between exploration and
+exploitation, i.e., how many trials are explored versus how long a
+single trial is trained for. Because the right tradeoff between
+exploration and exploitation is hard to know in advance, the
+``adaptive_asha`` algorithm tries several SHA searches with different
+tradeoffs.
+
+The configuration settings available to Determined experiments running
+in ``adaptive_asha`` mode mostly affect the ASHA subroutines directly.
+The ``mode`` configuration is the only one affecting the decisions of
+the ``adaptive_asha`` searcher, by changing the number and types of ASHA
+subroutines spawned.
+
+The first section here gives a description of ASHA. The second section
+describes why you would choose ``adaptive_asha`` over plain
+``asynchronous_halving``. The third section gives a summary of the
+asynchronous part of ``adaptive_asha``. The finally section and
+conclusion is a set of FAQs regarding ``adaptive_asha``.
+
+ASHA
+====
+
+At a high level, ASHA prunes ("halves") a set of trials in successive
+rounds we call *rungs*. ASHA starts with an initial set of trials. (A
+trial means one model, with a fixed set of hyperparameter values.) ASHA
+trains all the trials for some length and the trials with the worst
+validation performance are discarded. In the next rung, the remaining
+trials are trained for a longer period of time, and then trials with the
+worst validation performance are pruned once again. This is repeated
+until the maximum training length is reached.
+
+First, an example of ASHA.
+
+-  Rung 1: ASHA creates N initial trials; the hyperparameter values for
+   each trial are randomly sampled from the hyperparameters defined in
+   the experiment configuration file. Each trial is trained for 1 epoch,
+   and then validation metrics are computed.
+
+-  Rung 2: ASHA picks the N/4 top-performing trials according to
+   validation metrics. These are trained for 4 epochs.
+
+-  Rung 3: ASHA picks the N/16 top-performing trials according to
+   validation metrics. These are trained for 16 epochs.
+
+At the end, the trial with best performance has the hyperparameter
+setting the ASHA searcher returns.
+
+In the example above, ``divisor`` is 4, which determines what fraction
```

Comment:
Proposed edit:
In the example above, we generalize "halving" with a field called ``divisor``, which determines what fraction of trials are kept in successive rungs, as well as the training length
in successive rungs.